docs/guides/mainnet-forking.md

```diff
@@ -0,0 +1,122 @@
+# Mainnet forking
+
+You can start an instance of the Hardhat Network that forks the mainnet. This means that it will simulate having the same state as the mainnet, but it will work as a local development node. That way you can interact with deployed protocols and test complex interactions without the risk of doing it in the real mainnet.
+
+To use this feature, the only thing you need is the URL of a node to connect to. This can be an [Infura](https://infura.io/) or [Alchemy](https://alchemyapi.io/) node, or a local one. If you don't know which to use, we recommend you use [Alchemy](https://alchemyapi.io/) 
+
+## Forking from mainnet
+
+The easiest way to test this feature is to start a node from the command line:
+
+```
+npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<key>
+```
+
+Here you have to replace `<key>` with your Alchemy key. If you are using Infura, you need a different URL, and a project id instead of a key.
+
+You can also configure Hardhat to always do this, by setting it in the config:
+
+```js
+networks: {
+  hardhat: {
+    forking: {
+      url: "https://eth-mainnet.alchemyapi.io/v2/<key>"
+    }
+  }
+}
+```
+
+## Pinning a block
+
+If you start a forked node the way we explained in the previous section, Hardhat Network will fork the mainnet from the latest block available at that moment. This has some problems.
+
+First, it's not reproducible: you might run it at some point and it will use the block 11095000, and then run it one hour later and it will use the block 11095240. In that interval of time, the state of the blockchain will change, and this could cause your tests or scripts to behave differently.
```

Comment:
```suggestion
There are two reasons for this:
- The state your tests run against may change between runs. This could cause your tests or scripts to behave differently.
```